src/Compilers/CSharp/Portable/Symbols/Symbol.cs

```diff
@@ -1157,6 +1156,88 @@ public ImmutableArray<SymbolDisplayPart> ToDisplayParts(SymbolDisplayFormat form
             }
         }
 
+        internal byte? GetNullableContextValue()
+        {
+            var symbol = this;
+            do
+            {
+                var value = symbol.GetLocalNullableContextValue();
+                if (value != null)
+                {
+                    return value;
+                }
+                symbol = symbol.ContainingSymbol;
+            } while (!(symbol is null));
+            return null;
+        }
+
+        internal virtual byte? GetLocalNullableContextValue()
+        {
+            return null;
+        }
+
+        internal void GetCommonNullableValues(CSharpCompilation compilation, ref MostCommonNullableValueBuilder builder)
+        {
+            switch (this.Kind)
+            {
+                case SymbolKind.NamedType:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        builder.AddValue(this.GetLocalNullableContextValue());
+                    }
+                    break;
+                case SymbolKind.Event:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        builder.AddValue(((EventSymbol)this).TypeWithAnnotations);
+                    }
+                    break;
+                case SymbolKind.Field:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        builder.AddValue(((FieldSymbol)this).TypeWithAnnotations);
+                    }
+                    break;
+                case SymbolKind.Method:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        builder.AddValue(this.GetLocalNullableContextValue());
+                    }
+                    break;
+                case SymbolKind.Property:
+                    if (compilation.ShouldEmitNullableAttributes(this))
+                    {
+                        var property = (PropertySymbol)this;
+                        builder.AddValue(property.TypeWithAnnotations);
```

Comment:
>builder.AddValue(property.TypeWithAnnotations); [](start = 24, length = 47)

It feels like for the more accurate count we should delegate to accessors. And, if this method will be called for them anyway, we should do nothing here.